### PR TITLE
fatal commit

### PR DESCRIPTION
**Pull request by AlexForJava**

### CODE REVIEW

File: src/com/gmail/chernii/oleksii/Main.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import com.gmail.chernii.oleksii.list.MyLinkedList;
+import com.gmail.chernii.oleksii.stack.MyStack;
+import com.gmail.chernii.oleksii.stack.Stack;
+
+/**
+ * Created by Space on 05.03.2019.
+ */
+public class Main {
+    public static void main(String[] args) {
+        //testLinkedList();
```
Comment (Anderwerty) [inline src/com/gmail/chernii/oleksii/Main.java:14]: Do you need this code?

File: src/com/gmail/chernii/oleksii/list/MyArrayList.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * Created by Space on 05.03.2019.
+ */
+public class MyArrayList<E> implements List<E> {
+    private E[] elements;
+    private int length;
+    private int size;
+
+    public MyArrayList() {
+        this.length = 10;
```
Comment (Anderwerty) [inline src/com/gmail/chernii/oleksii/list/MyArrayList.java:16]: `10` looks like a magic number. Please use a constant.

File: src/com/gmail/chernii/oleksii/list/MyArrayList.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            length *= 2;
+            elements = Arrays.copyOf(elements, length);
+            elements[size++] = item;
+        }
+    }
+
+    @Override
+    public E get(int index) {
+        if (index < 0 || index > size - 1 || size == 0) {
+            return null;
```
Comment (Anderwerty) [inline src/com/gmail/chernii/oleksii/list/MyArrayList.java:35]: You arrayList could contain a null element. So, we could find if get return as null is it the element or not right index.  
Better to throw a custom exception.

File: src/com/gmail/chernii/oleksii/list/MyLinkedList.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            first = newNode;
+        } else {
+            tempLast.next = newNode;
+        }
+        size++;
+    }
+
+    @Override
+    public E get(int index) {
+        return (!checkIndex(index)) ? node(index).item : null;
```
Comment (Anderwerty) [inline src/com/gmail/chernii/oleksii/list/MyLinkedList.java:34]: The same

File: src/com/gmail/chernii/oleksii/list/MyLinkedList.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        }
+        node.item = null;
+        size--;
+    }
+
+    private Node<E> node(int index) {
+        if (index < (size / 2)) {
+            Node<E> temp = this.first;
+            for (int i = 0; i < index; i++)
+                temp = temp.next;
```
Comment (Anderwerty) [inline src/com/gmail/chernii/oleksii/list/MyLinkedList.java:78]: Add braces
Braces are used with if, else, for, do and while statements, even when the body is empty or contains only a single statement.
https://mate-academy.github.io/style-guides/java/java.html#s4.1.1-braces-always-used

File: src/com/gmail/chernii/oleksii/list/MyLinkedList.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+            return temp;
+        } else {
+            Node<E> temp = last;
+            for (int i = size - 1; i > index; i--)
+                temp = temp.prev;
```
Comment (Anderwerty) [inline src/com/gmail/chernii/oleksii/list/MyLinkedList.java:83]: Braces

**Review by Anderwerty — CHANGES_REQUESTED** [review]
Please fix comments

**Comment by AlexForJava** [comment]
Ok, thanks 

5 марта 2019, 13:19:05, от "Алексей" <hqq@ukr.net>:

@Anderwerty requested changes on this pull request.

Please fix comments

In src/com/gmail/chernii/oleksii/Main.java:

> @@ -0,0 +1,80 @@
+package com.gmail.chernii.oleksii;
+
+import com.gmail.chernii.oleksii.list.List;
+import com.gmail.chernii.oleksii.list.MyArrayList;
+import com.gmail.chernii.oleksii.list.MyLinkedList;
+import com.gmail.chernii.oleksii.stack.MyStack;
+import com.gmail.chernii.oleksii.stack.Stack;
+
+/**
+ * Created by Space on 05.03.2019.
+ */
+public class Main {
+    public static void main(String[] args) {
+        //testLinkedList();
Do you need this code?

In src/com/gmail/chernii/oleksii/list/MyArrayList.java:

> +package com.gmail.chernii.oleksii.list;
+
+import com.gmail.chernii.oleksii.list.List;
+
+import java.util.Arrays;
+
+/**
+ * Created by Space on 05.03.2019.
+ */
+public class MyArrayList<E> implements List<E> {
+    private E[] elements;
+    private int length;
+    private int size;
+
+    public MyArrayList() {
+        this.length = 10;
10 looks like a magic number. Please use a constant.

In src/com/gmail/chernii/oleksii/list/MyArrayList.java:

> +
+    @Override
+    public void add(E item) {
+        if (size < length - 1) {
+            elements[size++] = item;
+        } else {
+            length *= 2;
+            elements = Arrays.copyOf(elements, length);
+            elements[size++] = item;
+        }
+    }
+
+    @Override
+    public E get(int index) {
+        if (index < 0 || index > size - 1 || size == 0) {
+            return null;
You arrayList could contain a null element. So, we could find if get return as null is it the element or not right index.
Better to throw a custom exception.

In src/com/gmail/chernii/oleksii/list/MyLinkedList.java:

> +    @Override
+    public void add(E item) {
+        Node<E> tempLast = last;
+        Node<E> newNode = new Node<E>(tempLast, item, null);
+        last = newNode;
+        if (tempLast == null) {
+            first = newNode;
+        } else {
+            tempLast.next = newNode;
+        }
+        size++;
+    }
+
+    @Override
+    public E get(int index) {
+        return (!checkIndex(index)) ? node(index).item : null;
The same

In src/com/gmail/chernii/oleksii/list/MyLinkedList.java:

> +        }
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            node.next = null;
+        }
+        node.item = null;
+        size--;
+    }
+
+    private Node<E> node(int index) {
+        if (index < (size / 2)) {
+            Node<E> temp = this.first;
+            for (int i = 0; i < index; i++)
+                temp = temp.next;
Add braces
Braces are used with if, else, for, do and while statements, even when the body is empty or contains only a single statement.
https://mate-academy.github.io/style-guides/java/java.html#s4.1.1-braces-always-used

In src/com/gmail/chernii/oleksii/list/MyLinkedList.java:

> +            node.next = null;
+        }
+        node.item = null;
+        size--;
+    }
+
+    private Node<E> node(int index) {
+        if (index < (size / 2)) {
+            Node<E> temp = this.first;
+            for (int i = 0; i < index; i++)
+                temp = temp.next;
+            return temp;
+        } else {
+            Node<E> temp = last;
+            for (int i = size - 1; i > index; i--)
+                temp = temp.prev;
Braces

**Pull request closed without merging**